mkosi/__init__.py

```diff
@@ -5584,24 +5655,87 @@ def make_build_dir(args: CommandLineArguments) -> None:
     mkdir_last(args.build_dir, 0o755)
 
 
+def setup_ssh(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> Optional[TextIO]:
+    if do_run_build_script or for_cache or not args.ssh:
+        return None
+
+    if args.distribution in (Distribution.debian, Distribution.ubuntu):
+        unit = "ssh"
+    else:
+        unit = "sshd"
+
+    run(["systemctl", "--root", root, "enable", unit])
+
+    f: TextIO = cast(
+        TextIO,
+        tempfile.NamedTemporaryFile(
+            mode="w+", prefix=".mkosi-", encoding="utf-8", dir=os.path.dirname(args.output_sshkey)
+        ),
+    )
+
+    with complete_step("Generating SSH keypair"):
+        # Write a 'y' to confirm to overwrite the file.
+        run(["ssh-keygen", "-f", f.name, "-N", args.password or ""], input=f"y\n", text=True)
```

Comment:
Why exactly is the input needed here? Just running `ssh-keygen -f ./testkey -N foobar` I'm not asked for input. 

Also, I think it would make sense to add a `-C mkosi` (or similar) so that the comment for the host key isn't `builduser@buildhost`. 

Another minor nit, this leaves open which algorithm is used and for backwards compat it still defaults to RSA. In the spirit of modern defaults, we could choose `-t ed25519` here, which has been an option for years already, so should be supported everywhere.

Reply:
We need to overwrite the named temporary file we create just before running ssh-keygen. I did it this way so that we can still take advantage of the named temporary file being automatically removed if it goes out of scope so that we don't leave anything to clean up if we fail somwhere further down the line.